server/sonar-web/src/main/js/apps/quality-gates/components/Projects.js

```diff
@@ -18,55 +18,116 @@
+    projects: []
+  };
+
   componentDidMount() {
-    this.renderSelectList();
+    this.handleSearch('', 'selected');
   }
 
-  renderSelectList = () => {
-    if (!this.container) return;
+  handleSearch = (query /*: string*/, selected /*: string */) => {
+    const { qualityGate, organization } = this.props;
+    const requestData = {
+      gateId: qualityGate.id,
+      pageSize: 100,
+      selected
+    };
+
+    if (query !== '') {
+      requestData.query = query;
+    }
+
+    if (organization) {
+      requestData.organization = organization;
+    }
+
+    return searchGates(requestData).then(
+      data => {
+        this.setState({
+          projects: data.results.map(project => {
+            return { key: project.id, name: project.name, selected: project.selected };
+          })
+        });
+      },
+      () => {}
+    );
+  };
+
+  handleSelect = (key /*: number|string*/) => {
+    const { qualityGate, organization } = this.props;
+    const requestData = {
+      gateId: qualityGate.id,
+      projectId: parseInt(key, 10)
+    };
+
+    if (organization) {
+      requestData.organization = organization;
+    }
+
+    return associateGateWithProject(requestData).then(
+      () => {
+        this.setState(state => {
+          const projects = state.projects.map(project => {
+            return project.key === key ? { ...project, selected: true } : project;
+          });
+          return { projects };
+        });
+      },
+      () => {}
+    );
+  };
 
-    const { qualityGate, edit, organization } = this.props;
+  handleUnselect = (key /*: number|string*/) => {
+    const { qualityGate, organization } = this.props;
+    const requestData = {
+      gateId: qualityGate.id,
+      projectId: parseInt(key, 10)
+    };
 
-    const extra = { gateId: qualityGate.id };
-    let orgQuery = '';
     if (organization) {
-      extra.organization = organization;
-      orgQuery = '&organization=' + organization;
+      requestData.organization = organization;
     }
 
-    // eslint-disable-next-line no-new
-    new SelectList({
-      el: this.container,
-      width: '100%',
-      readOnly: !edit,
-      focusSearch: false,
-      dangerouslyUnescapedHtmlFormat: item => escapeHtml(item.name),
-      searchUrl: getBaseUrl() + `/api/qualitygates/search?gateId=${qualityGate.id}${orgQuery}`,
-      selectUrl: getBaseUrl() + '/api/qualitygates/select',
-      deselectUrl: getBaseUrl() + '/api/qualitygates/deselect',
-      extra,
-      selectParameter: 'projectId',
-      selectParameterValue: 'id',
-      labels: {
-        selected: translate('quality_gates.projects.with'),
-        deselected: translate('quality_gates.projects.without'),
-        all: translate('quality_gates.projects.all'),
-        noResults: translate('quality_gates.projects.noResults')
+    return dissociateGateWithProject(requestData).then(
+      () => {
+        this.setState(state => {
+          const projects = state.projects.map(project => {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove curly braces and "return" from this arrow function body. ([javascript:S3524](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3524))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&pullRequest=3152&issues=AWIkUgyQ7DpOGbPQ2tAD&open=AWIkUgyQ7DpOGbPQ2tAD)

[ISSUE_KEY]: AWIkUgyQ7DpOGbPQ2tAD